feat: add Sky Sync

sicsix opened the pull request:
- Synchronises lighting and volumetric lighting to the sun and moons
- Sun is now visible as it crosses the horizon
- Fix for sun appearing higher in the sky based on player altitude, sunrise now the same at all heights
- Optional alternate sun path that travels through the southern sky
- Intensity adjustment of lighting and VL based on moon phase, supports Moon and Stars
- Sun is the dominant light source, but at night the moons can be selected to switch smoothly based on brightest or fixed to Masser or Secunda
- Light source is clamped slightly above horizon and fades out based on visibility of the sun/moon to prevent shenanigans
- Fixes sky orientation in interiors

sicsix committed:
Initial commit

sicsix committed:
style: 🎨 apply clang-format changes

alandtse commented on src/Features/SkySync.cpp:
```diff
@@ -0,0 +1,433 @@
+	}
+
+	const float time = sky->currentGameHour;
+	const bool isDayTime = time > timings.sunriseFadeOutMoonEnd && time < timings.sunsetFadeInMoonStart;
```
Was the built in function not appropriate?

sicsix replied:
Yeah, this calculation differs slightly. It determines when the sun/moons take lighting priority rather whether it is within daylight hours.

alandtse commented on src/Features/SkySync.h:
```diff
@@ -0,0 +1,175 @@
+	virtual void SaveSettings(json& o_json) override;
+	virtual void RestoreDefaultSettings() override;
+
+	virtual bool SupportsVR() override { return true; }
```
Let's also make sure VR Address Library is set to 0.175.0. We've missed that in the last few prs. https://github.com/doodlum/skyrim-community-shaders/blob/c188df38a9ff0ac03344e827182ce1840b0256ca/src/XSEPlugin.cpp#L156

sicsix replied:
Ah cheers didn't know this was a thing, updated in latest.

sicsix committed:
updated VR Address Library min version

alandtse committed:
feat: detect DVLaSS

This moves the hooks to data loaded. It appears to still work fine on my
side, but please confirm.

alandtse committed:
style: 🎨 apply clang-format changes

alandtse commented:
I have a PR into you for this too. https://github.com/sicsix/skyrim-community-shaders/pull/1

Another optional improvement would be when Brightest is selected, that you actually list which body is currently brightest. Or even just list all 3 items and provide brightest values in the menu so it's very obvious which is being selected.

sicsix committed:
feat: detect DVLaSS

sicsix committed:
fix volumetric lighting being reduced too much when intensity below 1

sicsix committed:
fix crash on hook

alandtse commented on src/Features/SkySync.cpp:
```diff
@@ -0,0 +1,441 @@
+		disableDueToConflict("EVLaS");
+}
+
+void SkySync::PostPostLoad()
```
Hooking this now means that it doesn't actually disable.

sicsix replied:
I've modified the logic in latest so that DLL conflicts are now checked in PostPostLoad then hooks are applied after. ESP conflicts are checked in DataLoaded after mods have loaded - the hooks will already be in place at this point but the module will self-disable and show up as an issue in the UI.

This should fix the intermittent crashing for some users when hooks are applied in DataLoaded.

sicsix committed:
fixed conflict detection and automatic disabling

sicsix committed:
style: 🎨 apply clang-format changes

doodlum commented:
Is this ready to merge? I can see there have been changes since the review

alandtse commented:
Yes from my end.
